workspaces/src/worker/mod.rs

```diff
@@ -13,6 +14,7 @@ use crate::{Network, Result};
 /// deploying a contract, or interacting with transactions.
 pub struct Worker<T: ?Sized> {
     pub(crate) workspace: Arc<T>,
+    pub(crate) on_transact: Vec<GasHook>,
```

Comment:
This is now a vector of hooks, so we should rename it to be plural, something like `tx_callbacks`